place_tracker/lib/place_tracker_app.dart

```diff
@@ -57,80 +50,58 @@ class _PlaceTrackerHomePage extends StatelessWidget {
             padding: EdgeInsets.fromLTRB(0.0, 0.0, 16.0, 0.0),
             child: IconButton(
               icon: Icon(
-                AppState.of(context).viewType == PlaceTrackerViewType.map
+                state.viewType == PlaceTrackerViewType.map
                     ? Icons.list
                     : Icons.map,
                 size: 32.0,
               ),
               onPressed: () {
-                AppState.updateWith(
-                  context,
-                  viewType:
-                      AppState.of(context).viewType == PlaceTrackerViewType.map
-                          ? PlaceTrackerViewType.list
-                          : PlaceTrackerViewType.map,
+                state.changeViewType(
+                  state.viewType == PlaceTrackerViewType.map
+                      ? PlaceTrackerViewType.list
+                      : PlaceTrackerViewType.map,
                 );
               },
             ),
           ),
         ],
       ),
       body: IndexedStack(
-        index:
-            AppState.of(context).viewType == PlaceTrackerViewType.map ? 0 : 1,
+        index: state.viewType == PlaceTrackerViewType.map ? 0 : 1,
         children: <Widget>[
           PlaceMap(center: const LatLng(45.521563, -122.677433)),
-          PlaceList(),
+          PlaceList()
         ],
       ),
     );
   }
 }
 
-class AppState {
-  const AppState({
+class AppState with ChangeNotifier {
+  AppState({
     this.places = StubData.places,
     this.selectedCategory = PlaceCategory.favorite,
     this.viewType = PlaceTrackerViewType.map,
   })  : assert(places != null),
         assert(selectedCategory != null);
 
-  final List<Place> places;
-  final PlaceCategory selectedCategory;
-  final PlaceTrackerViewType viewType;
+  List<Place> places;
+  PlaceCategory selectedCategory;
+  PlaceTrackerViewType viewType;
 
-  AppState copyWith({
-    List<Place> places,
-    PlaceCategory selectedCategory,
-    PlaceTrackerViewType viewType,
-  }) {
-    return AppState(
-      places: places ?? this.places,
-      selectedCategory: selectedCategory ?? this.selectedCategory,
-      viewType: viewType ?? this.viewType,
-    );
+  void changeViewType(PlaceTrackerViewType viewType) {
```

Comment:
This is also a style convention, but `setViewType`, `setSelectedCategory`, etc. would likely be better names here since these methods are essentially setters.